### ISSUE TITLE
Refactor objects inheritance to simplify API usage

### ISSUE THREAD

**Issue by vtexier** [duniter/duniter-python-api#28]
The inheritance pattern use in the API can be simplified.

For example, in a request like : 

``` python

bma.node.Summary(BMAEndpoint.from_inline(BMA_ENDPOINT).conn_handler()).get(AIOHTTP_SESSION)
```
- The endpoint and the aiohttp session could be passed together in one parameter as a connexion object.
- The get() part can be removed.

``` python
conn_handler = ConnexionHandler(AIOHTTP_SESSION, BMAEndpoint.from_inline(BMA_ENDPOINT))
bma.node.Summary(conn_handler)
```

**Comment by vtexier** [comment]
First draft: 

Before...
```python
    AIOHTTP_SESSION = aiohttp.ClientSession()
    connection = BMAEndpoint.from_inline(BMA_ENDPOINT).conn_handler()

    return await bma.node.Summary(connection).get(AIOHTTP_SESSION)
```
After:
```python
    AIOHTTP_SESSION = aiohttp.ClientSession()
    connection = BMAEndpoint.from_inline(BMA_ENDPOINT).conn_handler(AIOHTTP_SESSION)

    return await bma.node.Summary(connection).request()
```
What's new ?

* The `aiohttp.ClientSession()` instance is saved in the connection handler. So you don't have to specify it at each request.
* The `Summary` request handle the http method (GET here), you dont have to take care about it.
* The `__get__` method of the `API` class is deprecated. And it is a good thing cause PyCharm warns us about an error : the `async` keyword is forbidden on this method which is a reserved method of python classes.
* The `get` method of the `API` class is deprecated.
* If one day, the same request should be used with GET to read and POST to write, we can always use `requests_get` and `requests_put` methods.

What do you think about it ?

**Comment by vtexier** [comment]
**Second phase : replace modules classes by static methods with explicit arguments**

Refactored `GET block/number` and `POST block` is now simpler and well documented:

```python
URL_PATH = 'blockchain'

async def block(connection, number=0, block=None, signature=None):
    """
    GET/POST a block from/to the blockchain

    :param API.ConnectionHandler connection: Connection handler instance
    :param int number: Block number to get
    :param dict block: Block document to post
    :param str signature: Signature of the block document issuer
    :return: dict
    """

    client = API(connection, URL_PATH)
    # POST block
    if block is not None and signature is not None:
        r = await client.requests_post('/block', block=block, signature=signature)
        return r

    # GET block
    r = await client.requests_get('/block/%d' % number)
    return await client.parse_response(r, BLOCK_SCHEMA)
```

and request_data example is now:

```python

# connection handler from BMA endpoint
    connection = BMAEndpoint.from_inline(BMA_ENDPOINT).conn_handler(AIOHTTP_SESSION)

    # Get the node summary infos
    response = await bma.node.summary(connection)
    print(response)

    # Get the current block data
    response = await bma.blockchain.parameters(connection)
    print(response)

    # Get the current block data
    response = await bma.blockchain.current(connection)
    print(response)

    # Get the block number 10
    response = await bma.blockchain.block(connection, number=10)
    print(response)
```

**Comment by vtexier** [comment]
Branch dev merged in master at d6de5523a12cd64253c3035fe55e2b27b3157d71
